app/models/Aunthentication.js

```diff
@@ -0,0 +1,49 @@
+var mogoose = require('mongoose');
+var Schema = mongoose.Schema;
+var crypto = require('crypto');
+var jwt = require('jsonwebtoken');
+
+var AuthenticationSchema = new  Schema({
+    Email: {
+        type: string,
+        required: true,
+        unique: true
+    },
+    id: {
+        type : Number,
+        Required: true
+    },
+    username: {
+        type: String,
+        Required: true,
+        unique: true
+    },
+
+    hash: String,
+    salt: String
+});
+
+AuthenticationSchema.method.setPassword = function(password){
+    this.salt=crypto.randomBytes(16).toString('hex');
+    this.hash = crypto.pbkdf1Syc(password, this.salt, 1000,64, 'sha256').toString('hex');
+} ;
+
+AuthenticationSchema.method.validPassword = function(password){
+    var hash = crypto.pbkdf1Syn(password, this.salt,1000,64, 'sha256').tostring('hex');
+    return this.hash===hash;
+};
+
+AuthenticationSchama.method.generateJwt = function(){

```

Comment:
this method to generateJWT doesn't seem right to me
1. there is no name field so why are you encrypting it inside the jwt
2. its better to encode less information inside JWT it will decrease the size of JWT
3. my recommendation is to only use username and id to encrypt. Later on if we want more information about the user then we can just findById()
4. you need to set a secret to sign the jwt for better security

here is the code which you can refer
```javascript
UserSchema.methods.generateJWT = function () {
    var expiry = new Date();
    expiry.setDate(expiry.getDate() + 7);
    return jwt.sign({
        _id: this._id,
        username: this.username
    }, secret, {
        expiresIn: parseInt(expiry.getTime() / 1000)
    });
};
```